plugins/scaffolder-actions/scaffolder-utils/src/actions/sleep.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTemplateAction } from '@backstage/plugin-scaffolder-backend';
+import { InputError } from '@backstage/errors';
+
+export function createSleepAction() {
+    return createTemplateAction<{ amount: number }>({
+        id: "roadiehq:utils:sleep",
+        description: "Halts the scaffolding for the given amount of seconds",
```

Comment:
I just dont know about this one. It is kind of allowing someone to easily block a task worker. Can we at least limit the amount of seconds or maybe cause it to print a warning to the logs to say that it shouldnt be used in production or something..

Why is this needed?

Reply:
This works as an alternative to webhooks in many cases. Like setting up a Netlify page takes some time, and if you want to do an action after that page is done, you need some way of waiting until it is actually ready to accept the next steps

Reply:
Should there be a step for waiting on an asynchronous state instead of providing a sleep either for the specific purpose or in some generalized way?

Reply:
These probably should be that as well in addition but I don't think that is possible to do for all use cases.

Reply:
Yea, this is a general building block, we have the http action, and with this wait step you kinda can do anything. The general idea is in one template you create something and want to set something else on it, but the initial request returned 200 but it just need more time to actually be ready. 

Not sure if we can wait for async tasks otherwise. Limit can be an option, however this is intended for self hosted backstage too, and there I don't think why would someone screw up your own company with sleeping forever. If it is a pain for us I suggest we don't enable it on roadie.

Reply:
Cool, sounds good.